admin/controllers/helpdesk.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+const moment = require('moment-timezone')
+
+const checkWindowV2Service = require('../services/check-window-v2.service')
+const config = require('../config')
+const pupilRegisterService = require('../services/pupil-register.service')
+const schoolHomeFeatureEligibilityPresenter = require('../helpers/school-home-feature-eligibility-presenter')
+const schoolImpersonationService = require('../services/school-impersonation.service')
+const schoolService = require('../services/school.service')
+const resultPageAvailabilityService = require('../services/results-page-availability.service')
+const ValidationError = require('../lib/validation-error')
+
+const controller = {}
+
+/**
+ * Renders the helpdesk school impersonation page
+ * @param {object} req
+ * @param {object} res
+ * @param {function} next
+ * @param {object} error
+ * @returns {Promise.<void>}
+ */
+controller.getSchoolImpersonation = async (req, res, next, error = null) => {
+  res.locals.pageTitle = 'MTC Helpdesk School Impersonation'
+  schoolImpersonationService.removeImpersonation(req.user)
```

Comment:
@jon-shipley this is to ensure the school session data are wiped. It is not a mandatory step as the form itself will populate the session with new school data however I liked the idea of having a clean session object instead of replacing existing key values.

Reply:
Perhaps I wasn't clear.  You are changing state on the results of a GET which shouldn't happen.  You should do a POST to change state.

Reply:
also - what if they want to remain as that school?  the removal of impersonation needs to be explicit.

Reply:
@jon-shipley @GuyHarwood considering this indeed should have been an isolated action this has now changed and is an explicit POST route.